pipelines/ZeroShotClassification.go

```diff
@@ -0,0 +1,502 @@
+			output := &ZeroShotClassificationOutput{
+				Sequence: sequences[ind],
+			}
+
+			var entailmentLogits []float32
+			var contradictionLogits []float32
+
+			var entailmentID int
+			var contradictionID int
+			switch p.entailmentID {
+			case -1:
+				entailmentID = len(sequence[0]) - 1
+				contradictionID = 0
+			default:
+				entailmentID = p.entailmentID
+				contradictionID = 0
+				if entailmentID == 0 {
+					contradictionID = len(sequence[0]) - 1
+				}
+			}
+
+			for _, tensor := range sequence {
+				entailmentLogits = append(entailmentLogits, tensor[entailmentID])
+				contradictionLogits = append(contradictionLogits, tensor[contradictionID])
+			}
+
+			for i := range entailmentLogits {
+				logits := []float64{float64(contradictionLogits[i]), float64(entailmentLogits[i])}
+				exp_logits := []float64{math.Exp(logits[0]), math.Exp(logits[1])}
+				sum_exp_logits := exp_logits[0] + exp_logits[1]
+				score := exp_logits[1] / sum_exp_logits
+				LabelLikelihood[labels[i]] = score
+			}
+
+			// Define ss as a slice of anonymous structs
+			var ss []struct {
+				Key   string
+				Value float64
+			}
+			for k, v := range LabelLikelihood {
+				ss = append(ss, struct {
+					Key   string
+					Value float64
+				}{k, v})
+			}
+
+			// Sort the slice by the value field
+			sort.Slice(ss, func(i, j int) bool {
+				return ss[i].Value > ss[j].Value
+			})
+
+			output.SortedValues = ss
+			classificationOutputs = append(classificationOutputs, output)
+		}
+		return classificationOutputs, nil
+	} else {
+		for ind, sequence := range outputTensors {
+			output := &ZeroShotClassificationOutput{}
+
+			var entailmentLogits []float32
+			var entailmentID int
+			switch p.entailmentID {
+			case -1:
+				entailmentID = len(sequence[0]) - 1
+			default:
+				entailmentID = p.entailmentID
+			}
+			for _, tensor := range sequence {
+				entailmentLogits = append(entailmentLogits, tensor[entailmentID])
+			}
+
+			var numerator []float64
+			var logitSum float64
+			for _, logit := range entailmentLogits {
+				exp := math.Exp(float64(logit))
+				numerator = append(numerator, exp)
+				logitSum += exp
+			}
+
+			var quotient []float64
+
+			for ind, i := range numerator {
+				quotient = append(quotient, i/logitSum)
+				LabelLikelihood[labels[ind]] = quotient[ind]
+			}
+
+			output.Sequence = sequences[ind]
+
+			// Define ss as a slice of anonymous structs
+			var ss []struct {
+				Key   string
+				Value float64
+			}
+			for k, v := range LabelLikelihood {
+				ss = append(ss, struct {
+					Key   string
+					Value float64
+				}{k, v})
+			}
+
+			// Sort the slice by the value field
+			sort.Slice(ss, func(i, j int) bool {
+				return ss[i].Value > ss[j].Value
+			})
+
+			output.SortedValues = ss
+			classificationOutputs = append(classificationOutputs, output)
+		}
+
+		return classificationOutputs, nil
+	}
+
+}
+
+func (p *ZeroShotClassificationPipeline) RunPipeline() (ZeroShotClassificationOutputArr, error) {
```

Comment:
RunPipeline accepts []string (a sequence).
Labels and hypothesis template are initizialized at pipeline creation